warehouse/utils/utils.go

```diff
@@ -612,26 +621,36 @@ func Datatype(in interface{}) string {
 	return "string"
 }
 
-//ToSafeDBString to remove special characters
-func ToSafeDBString(provider string, str string) string {
-	res := ""
-	if str != "" {
-		r := []rune(str)
-		_, err := strconv.ParseInt(string(r[0]), 10, 64)
-		if err == nil {
-			str = "_" + str
+func ToSafeNamespace(provider string, name string) string {
+	var extractedValues []string
+	var extractedValue string
+	for _, c := range name {
+		asciiValue := int(c)
+		if (asciiValue >= 65 && asciiValue <= 90) || (asciiValue >= 97 && asciiValue <= 122) || (asciiValue >= 48 && asciiValue <= 57) {
+			extractedValue += string(c)
+		} else {
+			if extractedValue != "" {
+				extractedValues = append(extractedValues, extractedValue)
+			}
+			extractedValue = ""
 		}
-		regexForNotAlphaNumeric := regexp.MustCompile("[^a-zA-Z0-9_]+")
-		res = regexForNotAlphaNumeric.ReplaceAllString(str, "")
+	}
 
+	if extractedValue != "" {
+		extractedValues = append(extractedValues, extractedValue)
+	}
+	namespace := strings.Join(extractedValues, "_")
+	namespace = strcase.ToSnake(namespace)
+	if namespace != "" && int(namespace[0]) >= 48 && int(namespace[0]) <= 57 {
+		namespace = "_" + namespace
 	}
-	if res == "" {
-		res = "STRINGEMPTY"
+	if namespace == "" {
+		namespace = "stringempty"
 	}
-	if _, ok := ReservedKeywords[provider][strings.ToUpper(str)]; ok {
-		res = fmt.Sprintf(`_%s`, res)
+	if _, ok := ReservedKeywords[provider][strings.ToUpper(namespace)]; ok {
+		namespace = fmt.Sprintf(`_%s`, namespace)
 	}
-	return res
+	return misc.TruncateStr(namespace, 127)
 }
 
 func ToCase(provider string, str string) string {
```

Comment:
Name seems a bit confusing. `ToCase` --> to which case? Also, can this go into individual warehouse specific files through the interface?

Reply:
we are using this in multiple places like wh slave func claimAndProcess where we are not using any wh interface currently. Added comment for better readability

Reply:
Rename it ToProviderCase then?